services/src/main/scala/cromwell/services/keyvalue/impl/KeyValueServiceActor.scala

```diff
@@ -30,23 +31,13 @@ case class KeyValueServiceActor(serviceConfig: Config, globalConfig: Config) ext
     }
   }
 
-  private def doPut(put: KvPut): Future[KvResponse] = {
-    upsertExecutionInfo(
-      put.pair.key.workflowId,
-      put.pair.key.jobKey,
-      Map(put.pair.key.key -> put.pair.value)
-    ).map(_ => KvPutSuccess(put))
+  private def doPut(put: KvPut): Future[KvResponse] = Future.successful {
```

Comment:
Yup, I know how that works but look at the `receive` logic: Ignoring the specific implemention of `doPut`, the `receive` should not farm out state manipulation to something that returns a `Future`